src/App/Pages/Accounts/LoginSsoPageViewModel.cs

```diff
@@ -65,81 +69,96 @@ public async Task InitAsync()
+                    return;
+                }
 
-            await _deviceActionService.ShowLoadingAsync(AppResources.LoggingIn);
-            string ssoToken;
+                await _deviceActionService.ShowLoadingAsync(AppResources.LoggingIn);
 
-            try
-            {
                 var response = await _apiService.PreValidateSso(OrgIdentifier);
-                ssoToken = response.Token;
-            }
-            catch (ApiException e)
-            {
-                await _deviceActionService.HideLoadingAsync();
-                await _platformUtilsService.ShowDialogAsync(
-                    (e?.Error != null ? e.Error.GetSingleMessage() : AppResources.LoginSsoError),
-                    AppResources.AnErrorHasOccurred);
-                return;
-            }
 
-            var passwordOptions = new PasswordGenerationOptions(true);
-            passwordOptions.Length = 64;
+                if (response == null || string.IsNullOrWhiteSpace(response.Token))
+                {
+                    await _deviceActionService.HideLoadingAsync();
+                    await _platformUtilsService.ShowDialogAsync(AppResources.LoginSsoError);
+                    return;
+                }
 
-            var codeVerifier = await _passwordGenerationService.GeneratePasswordAsync(passwordOptions);
-            var codeVerifierHash = await _cryptoFunctionService.HashAsync(codeVerifier, CryptoHashAlgorithm.Sha256);
-            var codeChallenge = CoreHelpers.Base64UrlEncode(codeVerifierHash);
+                string ssoToken = response.Token;
 
-            var state = await _passwordGenerationService.GeneratePasswordAsync(passwordOptions);
 
-            var redirectUri = "bitwarden://sso-callback";
+                var passwordOptions = new PasswordGenerationOptions(true);
+                passwordOptions.Length = 64;
 
-            var url = _apiService.IdentityBaseUrl + "/connect/authorize?" +
-                      "client_id=" + _platformUtilsService.GetClientType().GetString() + "&" +
-                      "redirect_uri=" + Uri.EscapeDataString(redirectUri) + "&" +
-                      "response_type=code&scope=api%20offline_access&" +
-                      "state=" + state + "&code_challenge=" + codeChallenge + "&" +
-                      "code_challenge_method=S256&response_mode=query&" +
-                      "domain_hint=" + Uri.EscapeDataString(OrgIdentifier) + "&" +
-                      "ssoToken=" + Uri.EscapeDataString(ssoToken);
+                var codeVerifier = await _passwordGenerationService.GeneratePasswordAsync(passwordOptions);
+                var codeVerifierHash = await _cryptoFunctionService.HashAsync(codeVerifier, CryptoHashAlgorithm.Sha256);
+                var codeChallenge = CoreHelpers.Base64UrlEncode(codeVerifierHash);
+
+                var state = await _passwordGenerationService.GeneratePasswordAsync(passwordOptions);
+
+                var redirectUri = "bitwarden://sso-callback";
```

Comment:
I think this could be a `const`, I don't see it changing throughout the code. Maybe it can be extracted to a `const` in the class